.travis.yml

```diff
@@ -1,8 +1,23 @@
-language: java
-
 matrix:
   include:
+    - name: deploy markbind
+      language: node
+      node_js:
+        - '8'
+      install:
+        - npm i -g markbind-cli
+      script: cd docs && markbind build
+      deploy:
+        provider: pages
+        skip_cleanup: true
+        github_token: $GITHUB_TOKEN
+        keep_history: true
```

Comment:
may i ask if this is necessary?

Reply:
'keep_history' prevents from force-push at the gh-pages branch and hence protects the branch.